src/api/deploy/aws-elastic.ts

```diff
@@ -0,0 +1,252 @@
+                + ` --description "${description}" --auto-create-application`
+            };
+          }
+
+          return attachExec(container, command).toPromise();
+        })
+        .then(() => {
+          // 3. check if environment exists
+          return environmentExists(container, environmentName);
+        })
+        .then(exists => {
+          if (exists) {
+            // 4. create-environment
+            if (environmentTemplate) {
+              let command = {
+                type: CommandType.deploy, command: `aws elasticbeanstalk create-environment`
+                  + ` --application-name "${application}" --environment-name "${environmentName}"`
+                  + ` --template-name "${environmentTemplate}"`
+              };
+
+              return attachExec(container, command)
+                .toPromise()
+                .then(result => {
+                  if (!(result && result.data === 0)) {
+                    const msg = 'aws create environment failed';
+                    observer.next({ type: 'containerError', data: msg});
+                    return Promise.reject(1);
+                  }
+
+                  return Promise.resolve();
+                });
+            } else if (solutionStackName) {
+              let command = {
+                type: CommandType.deploy, command: `aws elasticbeanstalk create-environment`
+                  + ` --application-name "${application}" --environment-name "${environmentName}"`
+                  + ` --solution-stack-name "${solutionStackName}"`
+              };
+
+              return attachExec(container, command)
+                .toPromise()
+                .then(result => {
+                  if (!(result && result.data === 0)) {
+                    const msg = 'aws create environment failed';
+                    observer.next({ type: 'containerError', data: msg});
+                    return Promise.reject(1);
+                  }
+
+                  return Promise.resolve();
+                });
+            }
+
+            const msg = `Environment with name ${environmentName} doesn't exists, environment `
+              + `template-name or solution-stack-name has to be provided in deploy configuration`
+              + ` to successfully create new environment`;
+            observer.next({ type: 'containerError', data: msg});
+            return Promise.reject(1);
+          } else {
+            return Promise.resolve();
+          }
+        })
+        .then(() => {
+          let msg = style.yellow.open + style.bold.open + '==> deployment completed successfully!'
+            + style.bold.close + style.yellow.close + '\r\n';
+          observer.next({ type: 'data', data: msg });
+          observer.complete();
+        })
+        .catch(err => {
+          observer.error(err);
+          observer.complete();
+        });
+    } else {
+      observer.error(-1);
```

Comment:
still having this `-1` that we talked about.

Reply:
done